lib/Dialect/TritonGPU/Transforms/AccelerateMatmul.cpp

```diff
@@ -144,6 +145,29 @@ class BlockedToMMA : public mlir::RewritePattern {
     }
   }
 
+  unsigned getMmaV3InstrN(tt::DotOp dotOp, unsigned currN) const {
+    if (mmaV3InstrN > 0)
+      return mmaV3InstrN;
+
+    mmaV3InstrN = currN;
+    SetVector<Operation *> slices;
+    mlir::getForwardSlice(dotOp.getResult(), &slices);
+    auto iter =
+        llvm::find_if(slices, [](Operation *op) { return isa<tt::DotOp>(op); });
+
+    if (iter != slices.end()) {
+      tt::DotOp nextDotOp = dyn_cast<tt::DotOp>(*iter);
+      auto type = nextDotOp.getResult().getType().cast<RankedTensorType>();
+      auto AType = nextDotOp.getOperand(0).getType().cast<RankedTensorType>();
+      auto shapePerCTA = ttg::getShapePerCTA(type);
+      auto instrShape = mmaVersionToInstrShape(3, shapePerCTA, AType);
+
+      if (instrShape[1] < currN)
```

Comment:
if we have
```
%0 = dot ... : xxx -> MMAEncoding(instrShape[16, 128, 16])
%1 = convert_layout %0 : xxx -> SharedEncoding or other encoding
...
%9 = some other operations depend on %1
%10 = convert_layout % 9 : xxx -> MMAEncoding(instrShape[16, 64, 16])
%11 = dot %xx, %xx, %10 : xxx -> MMAEncoding(instrShape[16, 64, 16])
```
I suppose we should not change the instr shape of %0?

Reply:
actually， for perf consideration, I hope the instr shape to %0

and  I hope the operations between  %0 and %11 happens on mma encoding.